load-tests/src/load-test.ts

```diff
@@ -0,0 +1,282 @@
+    });
+
+    const data = res.json();
+    const { id } = data;
+    jar_id = id;
+
+    check(id, {
+      "valid id": (id) => Number.isInteger(id)
+    });
+
+    sleep(1);
+  });
+
+  group('create mission', () => {
+    const modelInput: MissionModelInsertInput = {
+      jar_id,
+      mission: "aerie_load-test" + rd,
+      name: "Banananation (load-test)"+rd,
+      version: "0.0.0"+ rd,
+    };
+
+    const data = req.hasura(gql.CREATE_MISSION_MODEL, {model: modelInput}, token, username);
+    check(data, {
+      "model id is valid": (data) => Number.isInteger(data.insert_mission_model_one.id)
+    });
+
+    sleep(5);
+  });
+
+}
+
+export function query_missions() {
+  const query = `#graphql
+    query {
+      mission_model {
+        id
+      }
+    }
+  `;
+
+  const data = req.hasura(query, {}, token, username);
+  check(data, {
+    "data is nonnull": (data) => data !== null,
+    "id is valid": (data) => Number.isInteger(data.mission_model[0].id)
+  });
+
+  sleep(Math.random() * 5);
+}
+
+export function insert_plans() {
+  const query = `#graphql
+    query {
+      mission_model {
+        id
+      }
+    }
+  `;
+
+  const rd = Math.random() * 10000;
+
+  const data = req.hasura(query, {}, token, username);
+  const { mission_model } = data;
+  const ids: number[] = mission_model;
+  const id = ids[Math.floor(Math.random() * ids.length)].id;
+
+  const input: CreatePlanInput = {
+    model_id: id,
+    name: 'test_plan' + rd + id,
+    start_time: plan_start_timestamp,
+    duration: "1d"
+  };
+
+  const resp = req.hasura(gql.CREATE_PLAN, {plan: input}, token, username);
+
+  sleep(Math.random() * 5);
+}
+
+export function insert_random_activities() {
+  const query = `#graphql
+    query {
+      plan {
+        id
+      }
+    }
+  `;
+
+  const rd = Math.random() * 10000;
+
+  const data = req.hasura(query, {}, token, username);
+  const { plan } = data;
+  const ids: number[] = plan;
+  const random_id = ids[Math.floor(Math.random() * ids.length)];
+
+  if (!random_id) {
+    sleep(5);
+    return;
+  }
+
+  const id = random_id.id;
+
+  const random_hour_offset = Math.floor(Math.random() * 24);
+
+  const input: ActivityInsertInput = {
+      arguments: {
+        biteSize: 1,
+      },
+      plan_id: id,
+      type: 'BiteBanana',
+      start_offset: random_hour_offset + 'h',
+  };
+
+  const resp = req.hasura(gql.CREATE_ACTIVITY_DIRECTIVE, {activityDirectiveInsertInput: input}, token, username);
+
+  sleep(Math.random() * 5);
+}
+
+export function insert_targeted_activities() {
+
+  const random_seconds_offset = Math.floor(Math.random() * 24 * 60 * 60);
+
+  const input: ActivityInsertInput = {
+      arguments: {
+        biteSize: 1,
+      },
+      plan_id: 1,
+      type: 'BiteBanana',
+      start_offset: random_seconds_offset + 's',
+  };
+
+  const resp = req.hasura(gql.CREATE_ACTIVITY_DIRECTIVE, {activityDirectiveInsertInput: input}, token, username);
+
+  sleep(Math.random() * 5);
```

Comment:
What are these `sleep` statements for?

Reply:
These have been removed, it was a misunderstanding of mine adding them in the first place, since the `k6` scheduler will automatically run iterations as fast as is defined in the `scenarios`.